src/zendoo/mcTestCall.cpp

```diff
@@ -141,10 +189,17 @@ void create_verify_test_cert_proof(std::string ps_type_raw, std::string cert_typ
         for(int i = 0; i < bt_list_length; i ++){
             backward_transfer_t bt;
 
-            assert(IsHex(argv[arg]));
+            std::vector<unsigned char> vchData;
+            assert(DecodeBase58(argv[arg++], vchData));
```

Comment:
Do we need [DecodeBase58Check](https://github.com/HorizenOfficial/zend_oo/blob/4eb908f4453ae9a1131676f4dc5cf81f4f46087a/src/base58.cpp#L124-L139) in addition to DecodeBase58? If the inputs are normal zen addresses they'd be not just Base58 encoded but they'd include a checksum.

Reply:
I think, for test tool is enough for now.
As you can mention `mcTestCall` duplicated the code of `DecodeBase58`. That's temporary until we solve the problem with the dep libs linkage, for some reason it raise errors. Then we will be able to use CBitcoinAddress with all the checks.